main.go

```diff
@@ -93,20 +101,81 @@ func main() {
 		return
 	}
 
-	println()
-	fmt.Printf("%s%s                                     %s\n", ansiCodes["invertOn"], ansiCodes["bold"], ansiCodes["reset"])
-	fmt.Printf("%s%s     io.finnet Key Recovery Tool     %s\n", ansiCodes["invertOn"], ansiCodes["bold"], ansiCodes["reset"])
-	fmt.Printf("%s%s               v3.1.1                %s\n", ansiCodes["invertOn"], ansiCodes["bold"], ansiCodes["reset"])
-	fmt.Printf("%s%s                                     %s\n", ansiCodes["invertOn"], ansiCodes["bold"], ansiCodes["reset"])
-	println()
+	fmt.Print(banner())
 
-	address, sk, _, err := runTool(files, vaultID, nonceOverride, quorumOverride, exportKSFile, passwordForKS)
+	appConfig := AppConfig{
+		filenames:      files,
+		nonceOverride:  *nonceOverride,
+		quorumOverride: *quorumOverride,
+		exportKSFile:   *exportKSFile,
+		passwordForKS:  *passwordForKS,
+	}
+
+	// First validate that files exist and are readable
+	if err := ValidateFiles(appConfig); err != nil {
+		fmt.Print(errorBox(err))
+		os.Exit(1)
+	}
+
+	/**
+	 * Run the steps to get the menmonics
+	 */
+	var vaultsDataFiles []VaultsDataFile = make([]VaultsDataFile, 0, len(appConfig.filenames))
+	f := NewMnemonicsForm(appConfig)
+	if err := f.Run(&vaultsDataFiles); err != nil {
+		fmt.Println(errorBox(err))
+		os.Exit(1)
+	}
```

Comment:
@notatestuser from what I've seen passing the to-write struct by reference is idiomatic go IIUC. Shall I make this a returned value?

Reply:
Personally I prefer a return so it's clear that there has been output, but structs can be mutated in the function like that yeah indeed

Reply:
Ok, I like that, I'll try

Reply:
applied feedback